### PR TITLE
Add com.geekbench.Geekbench6

### PR DESCRIPTION
**Pull request by razzeee**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

This seems to work fine, but now says it's running on the OS Flatpak, which should be fine.

I haven't reached out to them yet, but will send an email right after this.

I'm not sure if the icon is working fine like this. Github shows it weirdly and I couldn't get it to show in g-s.

### CODE REVIEW

**Comment by razzeee** [comment]
Email send

File: com.geekbench.Geekbench6.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - install -Dm755 geekbench/geekbench6 ${FLATPAK_DEST}/geekbench6
+      - install -Dm644 app-icon.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -Dm644 com.geekbench.Geekbench6.appdata.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.appdata.xml
+    sources:
+      - type: file
+        path: app-icon.svg
+      - type: file
+        path: com.geekbench.Geekbench6.appdata.xml
+      - type: archive
+        url: https://cdn.geekbench.com/Geekbench-6.1.0-Linux.tar.gz
```
Comment (hfiguiere) [inline com.geekbench.Geekbench6.yml:29]: how does it work on aarch64?

File: com.geekbench.Geekbench6.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+modules:
+  - name: geekbench
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/bin
+      - cp -r geekbench/* ${FLATPAK_DEST}/bin/
+      - install -Dm755 geekbench/geekbench6 ${FLATPAK_DEST}/geekbench6
+      - install -Dm644 app-icon.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - install -Dm644 com.geekbench.Geekbench6.appdata.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.appdata.xml
```
Comment (hfiguiere) [inline com.geekbench.Geekbench6.yml:22]: I don't see an installed desktop file.

File: com.geekbench.Geekbench6.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: geekbench6
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
```
Comment (hfiguiere) [inline com.geekbench.Geekbench6.yml:11]: does it really support wayland?

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:22]: I was unsure about that part, due to this being a terminal app and if having a desktop file would mean, it shows in the app drawer

File: com.geekbench.Geekbench6.yml (hunk 1 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:29]: I don't know and I don't think I have the means to test.

File: com.geekbench.Geekbench6.yml (hunk 3 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:11]: not sure, the benchmark itself never shows on screen. I figured giving this permission would be fine due to https://docs.flathub.org/docs/for-app-authors/requirements#unrestricted-permissions but we can just remove it.

**Comment by razzeee** [comment]
Compute benchmark run with this:
https://browser.geekbench.com/v6/cpu/1931686

Vulkan benchmark:
https://browser.geekbench.com/v6/compute/677002

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (hfiguiere) [reply com.geekbench.Geekbench6.yml:22]: you can mark desktop file as being terminal only

without it, no icon.

File: com.geekbench.Geekbench6.yml (hunk 3 above)
Comment (hfiguiere) [reply com.geekbench.Geekbench6.yml:11]: hence the question

File: com.geekbench.Geekbench6.yml (hunk 1 above)
Comment (hfiguiere) [reply com.geekbench.Geekbench6.yml:29]: unsually when some blob is shipped marked as "Linux" the chances it run on anything else but x86_64 are "unlikely". But still an aarch64 flatpak is built.

Comment (razzeee) [reply com.geekbench.Geekbench6.yml:29]: Safest is probably disabling. There are three binaries, from what I can tell.

`geekbench6`
`geekbench_avx2`
`geekbench_x86_64`

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:22]: Added a desktop file, but it shows in the app drawer now. Still seems to work fine, starting from there, but you don't get to pass any options.

File: com.geekbench.Geekbench6.yml (hunk 3 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:11]: Removed it and didn't seem to change the vulkan score (might have even improved it slightly, but that might be moot)

Comment (razzeee) [reply com.geekbench.Geekbench6.yml:11]: Maybe not as moot as I want too, probably need to run before after again, without working on the PC when it runs

https://browser.geekbench.com/v6/compute/compare/677002?baseline=678823

**Comment by hfiguiere** [comment]
bot, build com.geekbench.Geekbench6

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Stll require changes. notably the aarch64 with Intel binaries.

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (barthalion) [reply com.geekbench.Geekbench6.yml:21]: No point in adding a desktop file if users are expected to pass some params, but leaving the decision to you.

File: com.geekbench.Geekbench6.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - install -Dm644 com.geekbench.Geekbench6.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+    sources:
+      - type: file
+        path: app-icon.svg
+      - type: file
+        path: com.geekbench.Geekbench6.appdata.xml
+      - type: file
+        path: com.geekbench.Geekbench6.desktop
+      - type: archive
+        url: https://cdn.geekbench.com/Geekbench-6.1.0-Linux.tar.gz
```
Comment (barthalion) [inline com.geekbench.Geekbench6.yml:31]: Does the licence allow redistribution?

Comment (razzeee) [reply com.geekbench.Geekbench6.yml:31]: I couldn't find a license. Wanted to figure out if it works first.

I guess I should move this to extra data then?

**Comment by razzeee** [comment]
> Stll require changes. notably the aarch64 with Intel binaries.

You mean I should disable arm builds for this, right?

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:21]: So, how would I add a desktop file, but keep it from showing up outside of software/the website?

File: com.geekbench.Geekbench6.yml (hunk 4 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:35]: Using extra-data now

**Comment by hfiguiere** [comment]
bot, build com.geekbench.Geekbench6

**Comment by hfiguiere** [comment]
btw you are allowed to trigger builds.

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (hfiguiere) [reply com.geekbench.Geekbench6.yml:20]: `Terminal: true` should hide it from the start menu (or whatever system you DE uses)

File: com.geekbench.Geekbench6.yml (hunk 3 above)
Comment (hfiguiere) [reply com.geekbench.Geekbench6.yml:10]: fallback-x11 alone is not valid. It should be `x11` if it doesn't support wayland.

**Comment by hfiguiere** [comment]
is it ready otherwise?

File: com.geekbench.Geekbench6.yml (hunk 2 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:20]: hrm, I've set that in the .desktop file, as you've likely seen, but it still shows with gnome

File: com.geekbench.Geekbench6.yml (hunk 3 above)
Comment (razzeee) [reply com.geekbench.Geekbench6.yml:10]: Should be changed

**Comment by razzeee** [comment]
> is it ready otherwise?

I think so

**Comment by hfiguiere** [comment]
bot, build com.geekbench.Geekbench6

**Review by hfiguiere — APPROVED** [review]

**Pull request closed without merging**